File: SphrLibrary/Entities/OpenmHealth/OmhBodyEntityBase.cs
using System.Runtime.Serialization;

namespace SphrLibrary.Entities.OpenmHealth
{
    /// <summary>
    /// Open mHealth ボディ部リソースの基本クラスを表します。
    /// </summary>
    [DataContract()]
    [Serializable()]
    public abstract class OmhBodyEntityBase
    {
        [DataMember()]
        public TimeFrame effective_time_frame = new TimeFrame();

        public OmhBodyEntityBase() { }
    }
}

File: References/Java/SphrLibrary-Java/lib/src/main/java/sphrlibrary/entities/openmhealth/OmhBodyEntityBase.java
package sphrlibrary.entities.openmhealth;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

/**
 * Open mHealth の Body Entity の基底クラス
 */
@Data
@NoArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public abstract class OmhBodyEntityBase {

    /**
     * 取得時刻 or 期間
     */
    @NonNull
    private TimeFrame effectiveTimeFrame = new TimeFrame();

}
